experimental/ComplexReflectionGroups/src/ComplexReflectionGroupType.jl

```diff
@@ -0,0 +1,678 @@
+
+function is_spetsial(G::ComplexReflectionGroupType)
+    
+    # See Achar (2009)
+    excspet = [ 4, 6, 8, 14, 23, 24, 25, 26, 27, 28, 29, 30, 32, 33, 34, 35, 36, 37 ]
+
+    for t in G.type
+        if isa(t, Int)
+            if !(t in excspet)
+                return false
+            end
+        else
+            m = t[1]
+            p = t[2]
+            n = t[3]
+            if !( (p == 1) || (m == p) )
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+function num_reflection_classes(G::ComplexReflectionGroupType)
+
+    N = ZZ(0)
+
+    # Grabbed from Magma
+    excnum = [ 2, 4, 3, 5, 3, 4, 5, 6, 1, 2, 3, 4, 4, 5, 6, 7, 2, 3, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1 ]
+
+    for t in G.type
+        if isa(t, Int)
+            i = t
+            N += ZZ(excnum[i-3]) #index shift by 3 because exceptionals start with G4
+        else
+            (m,p,n) = t
+            
+            # See Thiel (2014), Theorem 15.27
+            if n > 2 || (n == 2 && is_odd(p))
+                N += div(ZZ(m),ZZ(p))
+            else
+                N += div(ZZ(m),ZZ(p)) + 1
+            end
+        end
+    end
+
+    return N
+
+end
+
+function center(G::ComplexReflectionGroupType)
+
+    orders = ZZRingElem[]
+
+    # The center of an irreducible group is always cyclic of order equal to the gcd
+    # of the degrees.
+    # See Thiel (2014), Proposition 13.32
+    for C in components(G)
+        append!(orders,gcd(degrees(C)))
+    end
+
+    return abelian_group(orders)
+
+end
+
+function field_of_definition(G::ComplexReflectionGroupType)
+    # Reference is Geck & Malle (2006)
+
+    fields = Field[]
+
+    for t in G.type
+        if isa(t, Int)
+            i = t
+            if i == 4
+                K,z = cyclotomic_field(3)
+            elseif i == 5
+                K,z = cyclotomic_field(3)
+            elseif i == 6
+                K,z = cyclotomic_field(12)
+            elseif i == 7
+                K,z = K,z = cyclotomic_field(12)
+            elseif i == 8
+                K,z = K,z = cyclotomic_field(4)
+            elseif i == 9
+                K,z = K,z = cyclotomic_field(8)
+            elseif i == 10
+                K,z = K,z = cyclotomic_field(12)
+            elseif i == 11
+                K,z = K,z = cyclotomic_field(24)
+            elseif i == 12
+                R,X = polynomial_ring(QQ, "X")
+                f = X^2 + 2
+                K,z = number_field(f)
+            elseif i == 13
+                K,z = K,z = cyclotomic_field(8)
+            elseif i == 14
+                R,X = polynomial_ring(QQ, "X")
+                f1 = X^3 - 1
+                f2 = X^2 + 2
+                K,(a,b) = number_field([f1,f2])
+            end
+            push!(fields, K)
+        else
+            m = ZZ(t[1])
+            p = ZZ(t[2])
+            n = ZZ(t[3])
+        end
+    end
+
+    if length(fields) == 1
+        return fields[1]
+    else
+        println(fields)
```

Comment:
Debugging leftover?
```suggestion
```

Reply:
Function commented out for the moment. Will remove when I finish the function (one day).